packages/dev/test/RelayServer.test.ts

```diff
@@ -605,6 +606,212 @@ contract('RelayServer', function (accounts: Truffle.Accounts) {
     })
   })
 
+  describe('withdrawToOwnerIfNeeded', function () {
+    let configFilename: string
+    let currentBlockNumber: number
+    const onceWithdrawConfig = JSON.stringify({
+      withdrawOnEthBalanceReached: 1e18,
+      leaveManagerWithAmountEth: 0.5e18,
+      repeat: false
+    })
+    beforeEach(async function () {
+      await env.newServerInstance({}, getTemporaryWorkdirs())
+      configFilename = `${env.relayServer.config.workdir}/withdraw.json`
+      assert.isFalse(fs.existsSync(configFilename))
+      currentBlockNumber = await env.web3.eth.getBlockNumber()
+    })
+    afterEach(async function () {
+      sinon.restore()
+    })
+    it('should not withdraw if relayer is not ready', async function () {
+      fs.writeFileSync(configFilename, onceWithdrawConfig)
+      env.relayServer.setReadyState(false)
+      const serverSpy = sinon.spy(env.relayServer)
+      const txHashes = await env.relayServer.withdrawToOwnerIfNeeded(currentBlockNumber)
+      assert.deepEqual(txHashes, [])
+      sinon.assert.calledOnce(serverSpy.isReady)
+      assert.isFalse(serverSpy.isReady.returnValues[0])
+    })
+
+    it('should not withdraw if missing config file', async function () {
+      const serverSpy = sinon.spy(env.relayServer)
+      const txHashes = await env.relayServer.withdrawToOwnerIfNeeded(currentBlockNumber)
+      assert.deepEqual(txHashes, [])
+      sinon.assert.calledOnce(serverSpy.isReady)
+      assert.isTrue(serverSpy.isReady.returnValues[0])
+    })
+
+    describe('_resolveWithdrawalConfig', function () {
+      async function assertReturnedError (message: string): Promise<void> {
+        const spy = sinon.spy(env.relayServer.logger, 'error')
+        await env.relayServer.withdrawToOwnerIfNeeded(currentBlockNumber)
+        sinon.assert.calledWith(spy, `withdrawToOwnerIfNeeded: ${message}`)
+        spy.restore()
+      }
+      it('should throw with bad config file', async function () {
```

Comment:
what is the net result ? the server stops on bad config? (maybe its not such a bad idea, since you don't have a way to signal the user the file is bad)

Reply:
No, the server doesn't stop, it logs an error on it.

Reply:
Not relevant anymore..